### PR TITLE
added SpringSecurity

### PR DESCRIPTION
**Pull request by Baldinini**

### CODE REVIEW

**Review by antoniukivan — APPROVED** [review]

File: src/main/java/mate/academy/annotation/EmailValidationConstrain.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package mate.academy.annotation;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class EmailValidationConstrain implements ConstraintValidator<EmailValidation, String> {
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/annotation/EmailValidationConstrain.java]:
```suggestion
public class EmailValidator implements ConstraintValidator<EmailValidation, String> {
```

File: src/main/java/mate/academy/annotation/EmailValidationConstrain.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    public static final String REGEX = "[a-zA-Z]+";
+
+    @Override
+    public void initialize(EmailValidation constraintAnnotation) {
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/annotation/EmailValidationConstrain.java]: we can remove this

File: src/main/java/mate/academy/annotation/PasswordValidationConstrain.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package mate.academy.annotation;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class PasswordValidationConstrain implements
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/annotation/PasswordValidationConstrain.java]:
```suggestion
public class PasswordValidator implements
```

File: src/main/java/mate/academy/annotation/PasswordValidationConstrain.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    private String repeatPassword;
+
+    public void initialize(PasswordValidation constraintAnnotation) {
+        this.password = constraintAnnotation.field();
+        this.repeatPassword = constraintAnnotation.fieldMatch();
+    }
+
+    public boolean isValid(Object value,
+                           ConstraintValidatorContext context) {
+
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/annotation/PasswordValidationConstrain.java]: redundant empty line

File: src/main/java/mate/academy/controllers/CustomGlobalExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/controllers/CustomGlobalExceptionHandler.java]: redundant empty line

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**